punchcut/src/lib.rs

```diff
@@ -0,0 +1,191 @@
+/*!
+Glyph loading.
+*/
+
+#![forbid(unsafe_code)]
+// TODO: this is temporary-- remove when hinting is added.
+#![allow(dead_code, unused_imports, unused_variables)]
+
+/// Re-export of peniko crate.
+pub use peniko;
+
+mod error;
+mod scaler;
+mod sink;
+
+pub mod source;
+
+/// Representations of fonts and font collections.
+pub mod font {
+    pub use read_fonts::{types::Tag, CollectionRef, FileRef, FontRef, TableProvider};
+}
+
+use font::Tag;
+use source::glyf;
+
+use core::str::FromStr;
+
+pub use error::{Error, Result};
+pub use scaler::{Scaler, ScalerBuilder};
+pub use sink::PathSink;
+
+/// Modes for hinting.
+///
+/// Only the `glyf` source supports all hinting modes.
+#[derive(Copy, Clone, PartialEq, Eq, Default, Debug)]
+#[allow(dead_code)]
+pub enum Hinting {
+    /// "Full" hinting mode. May generate rough outlines and poor horizontal
+    /// spacing.
+    Full,
+    /// Light hinting mode. This prevents most movement in the horizontal direction
+    /// with the exception of a per-font backward compatibility opt in.
+    Light,
+    /// Same as light, but with additional support for RGB subpixel rendering.
+    LightSubpixel,
+    /// Same as light subpixel, but always prevents adjustment in the horizontal
+    /// direction. This is the default mode.
+    #[default]
+    VerticalSubpixel,
+}
+
+/// Type for a normalized variation coordinate.
+pub type NormalizedCoord = read_fonts::types::F2Dot14;
+
+/// Type for a glyph identifier.
+pub type GlyphId = read_fonts::types::GlyphId;
+
+/// Setting for specifying a variation by tag and value.
```

Comment:
Useless comment, tell me something about it. What's it used for, what coordinate system is the value in, what thing(s) in the otspec should I use it to represent.

Reply:
This type will actually go away once the `feta` crate lands as it has a generic `Setting<T>` designed to support both variations and features. I'll make sure that one is better documented.